### PR TITLE
[Portstat]: Add -p option to support portstat timer

### PR DESCRIPTION
**Pull request by sihuihan88**

### CODE REVIEW

File: scripts/portstat [hunk 1]
```diff
@@ -261,13 +332,15 @@ Examples:
...
   portstat -r
+  portstat -p 20
 """)
     parser.add_argument('-c', '--clear', action='store_true', help='Copy & clear stats')
     parser.add_argument('-d', '--delete', action='store_true', help='Delete saved stats, either the uid or the specified tag')
     parser.add_argument('-D', '--delete-all', action='store_true', help='Delete all saved stats')
     parser.add_argument('-j', '--json', action='store_true', help='Display in JSON format')
     parser.add_argument('-r', '--raw', action='store_true', help='Raw stats (unmodified output of netstat)')
     parser.add_argument('-t', '--tag', type=str, help='Save stats with name TAG', default=None)
+    parser.add_argument('-p', '--period', type=int, help='Get the portstat within Period seconds', default=0)
```
Comment (jleveque) [inline scripts/portstat]: Make the help message clearer. Something like, "Display stats over a specified period (in seconds)."

File: scripts/portstat [hunk 2]
```diff
@@ -388,20 +401,25 @@ Examples:
...
+                print e.errno, e
         else:
-            cnstat_print(cnstat_dict, use_json)
+            if tag_name:
+                print "\nFile '%s' does not exist" % cnstat_fqn_file
+                print "Did you run 'portstat -c -t %s' to record the counters via tag %s?\n" % (tag_name, tag_name)
+            else:
+                cnstat_print(cnstat_dict, use_json)
+    else:
+        #wait for sometime and output the result
```
Comment (jleveque) [inline scripts/portstat]: "# Wait for the specified duration then gather new stats and output the difference"

File: scripts/portstat [hunk 3]
```diff
@@ -204,6 +205,76 @@ def parse_bcmcmd_ps(bcmcmd_output):
     bcmcmd_lines = bcmcmd_output.split("\n")
 
 
+def get_cnstat(cnstat_dir):
```
Comment (lguohan) [inline scripts/portstat]: it does not look the cnstat_dir is really used in this function, can you remove this argument?

File: scripts/portstat [hunk 4]
```diff
@@ -204,6 +205,76 @@ def parse_bcmcmd_ps(bcmcmd_output):
...
+        iface, stats = parsed[0]
+        proc[iface] = stats.split()
+
+    # At this point, either we'll create a file or open an existing one.
+    if not os.path.exists(cnstat_dir):
+        try:
+            os.makedirs(cnstat_dir)
+        except IOError as e:
+            print e.errno, e
+            sys.exit(1)
```
Comment (lguohan) [inline scripts/portstat]: here you only create a folder, but you do not use it anyway in the function. what's the purpose here? should we move it to some where else?

**Comment by lguohan** [comment]
can you make sure the existing portstat -c, portstat works

**Comment by jleveque** [comment]
@sihuihan88: I merged this PR. Please update the sonic-utilities submodule pointer in sonic-buildimage to reference this.

**Pull request merged**